src/toast/mpi.py

```diff
@@ -268,22 +269,28 @@ def exception_guard(comm=None):
     """
     log = Logger.get()
     failed = 0
+    rank = 0
+    if comm is not None:
+        rank = comm.rank
     try:
         yield
-    except:
-        msg = "Exception on process {}:\n".format(comm.rank)
+    except Exception:
```

Comment:
Bare `except` clauses should be avoided if possible. This is one of the few cases where it is warranted but perhaps it would pay to add a comment?